node/node.go

```diff
@@ -461,6 +461,17 @@ func makeNode(config *cfg.Config,
 		},
 	}
 
+	// this is a terrible hack, because typed nil interfaces are
```

Comment:
[no change requested] Granted it is kind of ugly, but it's not really a hack: A `nil` of pointer type and a `nil` of interface type are not equivalent, so it's entirely appropriate to not rely on the punning of the `nil` constant itself. (The fact that it's this way is certainly a wart on the language—but it's a long-standing and documented one, so probably the real lesson here is that we should avoid using pointers where we mean interfaces and vice versa 🙂)

Reply:
I deleted the word hack. I don't agree with the larger point.